src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
Seems fine to me if it makes the usage cleaner, though we could change the types to be `WebApplication` and `IWebApplicationInstance` if the problem is that the first class has the suffix `Builder`.

Consider adding 
```cs
        public WebApplicationBuilder UseConfiguration(
            string hostingJsonFile = null,
            string environmentVariablesPrefix = null,
            string[] commandLineArgs = null)
        {
            var configBuilder = new ConfigurationBuilder();

            if (!string.IsNullOrEmpty(hostingJsonFile))
            {
                configBuilder.AddJsonFile(
                    path: hostingJsonFile,
                    optional: true);
            }

            if (!string.IsNullOrEmpty(environmentVariablesPrefix))
            {
                configBuilder.AddEnvironmentVariables(
                    prefix: environmentVariablesPrefix);
            }

            if (commandLineArgs != null)
            {
                configBuilder.AddCommandLine(
                    args: commandLineArgs);
            }

            return UseConfiguration(configBuilder.Build());
        }
```
to enable the syntax
```cs
    public class StartupBlockingOnStart
    {
        // Entry point for the application.
        public static void Main(string[] args)
        {
            using (new WebApplicationBuilder()
                .UseConfiguration(commandLineArgs: args)
                .Start())
            {
                Console.ReadLine();
            }
        }
    }
```